runtime/src/threads/pthreads/threads-pthreads.c

```diff
@@ -81,6 +83,102 @@ static void            (*saved_threadEndFn)(void);
 static void*           initial_pthread_func(void*);
 static void*           pthread_func(void*);
 
+/* This conditional is necessary because on some operating systems
+ * (like OSX) the cstdlib malloc doesn't use nmap for the allocation
+ * and can't be mprotect-ed. In that case, we use the default pthread
+ * stack
+ */
+#ifndef CHPL_USING_CSTDLIB_MALLOC
```

Comment:
Probably I don't understand, but the problem here is the memory allocator (since cstdlib doesn't necessarily use mmap and mprotect can fail, like on OSX), not the eventual comm layer. I can agree that a runtime check is better than this kind of ifdef structure, however.